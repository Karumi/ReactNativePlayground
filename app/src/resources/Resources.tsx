import * as React from "react";
import { Image } from "react-native";
import asset from "../assets/asset";

class Resources extends React.Component {
    public render() {
        return (<Image source={asset("./images/arrow.png")} />);
    }
}

export default Resources;
