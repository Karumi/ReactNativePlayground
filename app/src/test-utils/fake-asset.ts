jest.mock("../assets/asset", () => {
return {
        default: (uri) => uri,
    };
});
