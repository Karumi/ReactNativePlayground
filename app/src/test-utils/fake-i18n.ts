
import { I18nResolver } from "i18n-ts";
import en from "../translations/en";

const i18n = { default: en };

jest.mock("../i18n", () => {
return {
        default: () => new I18nResolver(i18n,  "en").translation,
    };
} );
