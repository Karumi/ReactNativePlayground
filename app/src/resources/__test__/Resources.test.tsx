
import * as React from "react";
import * as renderer from "react-test-renderer";
import "../../test-utils/fake-asset";
import Resources from "../Resources";

it("renders correctly with defaults", () => {
    const toolbar = renderer.create(<Resources />).toJSON();
    expect(toolbar).toMatchSnapshot();
});
