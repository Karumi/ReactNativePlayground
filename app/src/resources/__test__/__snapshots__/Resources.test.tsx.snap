// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<Image
  source="./images/arrow.png"
/>
`;
